### PR TITLE
Use logback instead of log4j to fix logging issues

### PR DESCRIPTION
**Pull request by praste**
This fixes issue #221 . 
We have both `log4j` and `logback` jars and looks like `slf4j` is hitting `logback` jar  first(as classloader seems to be loading those before `log4j`). In this case `logback` ignore `log4j.xml` and uses default configuration where it spits out everything to console. Default console logger is synchronous and hence it could be performance issue as well if we are logging too much. 

`Logback` has really cool features like ability to change log level at runtime using JMX (which is available in `log4j2` but not in `log4j`)

I have put together `logback.xml` to mimic intending log configuration we were trying to achieve using `log4j.xml`, with some more added features like automatically delete log file more than 15 days old.

### CODE REVIEW

**Comment by gehel** [comment]
Looks good. And seeing the current state of logging in JmxTrans, it can only improve the situation ;-)

**Comment by gehel** [comment]
Thanks for your help !

File: jmxtrans.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ fi
 
 JAVA_HOME=${JAVA_HOME:-"/usr"}
 LOG_DIR=${LOG_DIR:-"."}
-LOG_FILE=${LOG_FILE:-"/dev/null"}
+STDOUT_REDIRECT_FILE=${STDOUT_REDIRECT_FILE:-"/dev/null"}
```
Comment (lookfirst) [inline jmxtrans.sh]: breaks backwards compat

Comment (gehel) [inline jmxtrans.sh]: Damn me ... I'll revert right now ...

**Comment by gehel** [comment]
Note that the move from log4j.xml to logback.xml is also a breaking compatibility, but it actually fixes the log4j / logback conflict that is breaking logging for a fairly long time ...

My opinion: we should break compatibility and make logging work ...

File: jmxtrans.sh (hunk 1 above)
Comment (praste) [inline jmxtrans.sh]: You don't really need a LOG_FILE variable as log file name was already
there in lo4j.xml

Using that environment variable was hack to make broken logging working.

Comment (gehel) [inline jmxtrans.sh]: Naming might not be the best ever, but it's not a hack, it's a good practice to capture stdout on a daemon process. You can think about it as 2 different levels of logging.

Comment (lookfirst) [inline jmxtrans.sh]: Agreed with @gehel

**Pull request merged**